metadata/metadata.go

```diff
@@ -482,6 +511,74 @@ func (resource *sourceVariantResource) Update(lookup ResourceLookup, updateRes R
 	return nil
 }
 
+func (resource *sourceVariantResource) IsEquivalent(other ResourceVariant) (bool, error) {
+	/**
+	Key Fields for a SourceVariant are
+	- Name
+	- Definition
+	- Owner
+	- Provider
+	*/
+	otherVariant, ok := other.(*sourceVariantResource)
+	if !ok {
+		return false, nil
+	}
+
+	thisProto := resource.serialized
+	otherProto := otherVariant.serialized
+
+	isDefinitionEqual := false
+	var err error
+	switch thisDef := thisProto.Definition.(type) {
+	case *pb.SourceVariant_Transformation:
+		if otherDef, ok := otherProto.Definition.(*pb.SourceVariant_Transformation); ok {
```

Comment:
we don't check for `!ok` here. Is the reason because `isDefinitionEqual` will be false so we will return `false, nil`?

Reply:
yeah, it'll just default to false